### DISCUSSION TITLE
New Release v4.10.4 - Added Home Page & Dashboard to Console App

### DISCUSSION THREAD

**Discussion by jongpie** [jongpie/NebulaLogger#500] [Announcements]
⚠️ **For orgs that are upgrading to this version & that use Logger Admin Dashboard plugin:** You will need to uninstall the Logger Admin Dashboard plugin first before you can upgrade to the core package in your org. The metadata from the Logger Admin Dashboard plugin has been added to the core package, and the plugin is now deprecated.

# Core Unlocked Package Changes

This release is focused on some related enhancements to help improve the usability & usefulness of the included lightning console app, including:
- Streamline access to some of Nebula Logger's features (namely, some of the included custom LWCs)
- Make it easier for admins & developers to find helpful resources for using/configuring Nebula Logger
- Provide admins & developers with a pre-built dashboard (and more reports) within the core package

These enhancements have culminated in a new home page, available in the console app

![image](https://github.com/jongpie/NebulaLogger/assets/1267157/56bd545d-8cbb-45aa-b3e9-f214551f9a90)

## Console app changes
- Changed label of console app from "Logger Console" to "Nebula Logger". The API name of the app, `LoggerConsole`, is not changing - this is just a label change.

  ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/a3dc8ce5-4d59-4218-90fb-596723736ec5)

- Added new home page `LoggerHomePage.flexipage-meta.xml` to the console app (configured as the app's default home page). It includes:
  - A new custom header component, `loggerHomeHeader`, that provides quick access to information about the current environment & quick access to the latest release notes for Nebula Logger
    ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/b0d04fec-a813-4260-aa2d-6e5844b482f4)

  - A tab set that includes 4 tabs:
    1. **Dashboard:** the first & default tab, this displays the new `LoggerAdmin` dashboard that is now included (previously, this dashboard was available as a separate plugin package)
    2. **Settings:** this displays the existing LWC `loggerSettings`, which was previously available via the custom tab "Logger Settings"
    3. **Event Stream:** this displays the existing LWC `logEntryEventStream`, which was previously available via the custom tab "Log Entry Event Stream"
    4. **Batch Purge:** this displays the existing LWC `logBatchPurge`, which was previously available via the custom tab "Log Batch Purge"
- Removed several custom tabs from the console app's navigation menu. Since the new home page displays several custom LWCs, their corresponding customs tabs are no longer included in the console app's navigation
  
  ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/a8b91b94-0b8f-40f2-925d-edda1220dfc5)

  - ⚠For orgs that use the unlocked package & are upgrading: by default, you will still see the old navigation options. You can either leave them as-is, or you can reset the tabs to use the new defaults. To reset & see the new navigation options, open the console app:
    1. In the console's navigation dropdown menu, select the "Edit" option at the bottom
    2. In the modal that opens, select the "Reset Navigation to Default" option at the bottom and click "Save"

    ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/32dc4a69-2269-4ddc-afe4-e2e431f25bca)

## Reports & Dashboard Changes
- The core package now includes a dashboard out of the box, `LoggerAdmin`. The dashboard is based on 7 new reports that are also included:
    - `LogEntryDailyRetentionSummary`
    - `LogEntryDailySummary`
    - `LogEntryDetails`
    - `LogEntryOriginSummary`
    - `LogEntryScenarioSummary`
    - `LogEntrySummary`
    - `LogEntryTagDetails`

## LWC Changes

- `logBatchPurge` LWC
  - Updated the LWC's config to make `isExposed = true` and added the target `lightning__HomePage` so that the component can be added in App Builder to the new home page
  - Added a confirmation prompt to the "Run Batch Purge" button to help prevent accidentally starting the job
  - Moved the spinner to only overlay on the `lightning-datatable` (when refreshing the data), instead of the spinner overlaying the entire page
  - Cleaned up some various code bits in the HTML and JS files (mostly to make some function names & terminology more consistent)

- `logEntryEventStream` LWC Enhancements
  - Updated the LWC's config to make `isExposed = true` and added the target `lightning__HomePage` so that the component can be added in App Builder to the new home page
  - Removed a line in `connectedCallback()` that was setting `document.title`
  - Minor CSS changes to improve the layout of the component within the home page's tab set

- `loggerSettings` LWC Enhancements
  - Updated the LWC's config to make `isExposed = true` and added the target `lightning__HomePage` so that the component can be added in App Builder to the new home page
  - Removed a line in `connectedCallback()` that was setting `document.title`

- `loggerHomeHeader` LWC - New!
  - Added new component that's used by the new home page. It provides a link to the current release's release notes, as well as a modal that provides environment details, using a combination of information from the `Organization` object and from making a callout to Salesforce's Status API, `https://api.status.salesforce.com/v1/instances/`.
    ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/5fac7189-5888-4186-9f17-c5184df788d6)

## Backend Changes
- Bugfix: fixed the instance method `LogEntryEventBuilder.setExceptionDetails(System.Exception apexException)` so it properly handles `null` values
- Bugfix: fixed the instance method `LogManagementDataSelector.getUsersByNameSearch(String searchTerm)` to use `LIMIT 20` so it doesn't hit a SOQL query limit in large orgs 😨 
- Added public method `Logger.callStatusApi()`, used to centralize some existing code that is used to retrieve additional orgs details from Salesforce's Status API, `https://api.status.salesforce.com/v1/instances/`. Note that this method is only intended to be used internally by Nebula Logger and could change in future releases.

# Logger Admin Dashboard Plugin - Now Deprecated! ⚠️
The Logger Admin Dashboard plugin is now deprecated, and will need to be uninstalled prior to upgrading to `v4.10.4`. The same dashboard & reports are now included in the core package, with one change: the dashboard is no longer a dynamic dashboard (due to the org limit of 5 dynamic dashboards). 

# Installation Info

## Core Unlocked Package - no namespace

Full Changelog: https://github.com/jongpie/NebulaLogger/compare/v4.10.3...v4.10.4
- SFDX CLI: `sfdx package install --wait 20 --security-type AdminsOnly --package 04t5Y0000023SC7QAM`
- Sandbox: https://test.salesforce.com/packaging/installPackage.apexp?p0=04t5Y0000023SC7QAM
- Production: https://login.salesforce.com/packaging/installPackage.apexp?p0=04t5Y0000023SC7QAM

<hr /><em>This discussion was created from the release <a href='https://github.com/jongpie/NebulaLogger/releases/tag/v4.10.4'>Added Home Page & Dashboard to Console App</a>.</em>

**Comment by jefersonchaves** [comment]
Wow, that is a superb release and a great addition, indeed. Seeing the peace still going is excellent, making it extremely useful!

**Comment by jamessimone** [comment]
Awesome! Looking forward to upgrading to take advantage of not having to null check on `setExceptionDetails` - the updated Nebula Logger app improvements are great!
